aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -462,22 +462,30 @@ impl AptosVM {
             let init_function = session.load_function(&module.self_id(), init_func_name, &[]);
             // it is ok to not have init_module function
             // init_module function should be (1) private and (2) has no return value
-            if init_function.is_ok() {
-                if verify_module_init_function(module).is_ok() {
-                    let args: Vec<Vec<u8>> = senders
-                        .iter()
-                        .map(|s| MoveValue::Signer(*s).simple_serialize().unwrap())
-                        .collect();
-                    session.execute_function_bypass_visibility(
-                        &module.self_id(),
-                        init_func_name,
-                        vec![],
-                        args,
-                        gas_meter,
-                    )?;
-                } else {
-                    return Err(PartialVMError::new(StatusCode::CONSTRAINT_NOT_SATISFIED)
-                        .finish(Location::Undefined));
+            match init_function {
+                Ok(_) => {
+                    if verify_module_init_function(module).is_ok() {
+                        let args: Vec<Vec<u8>> = senders
+                            .iter()
+                            .map(|s| MoveValue::Signer(*s).simple_serialize().unwrap())
+                            .collect();
+                        session.execute_function_bypass_visibility(
+                            &module.self_id(),
+                            init_func_name,
+                            vec![],
+                            args,
+                            gas_meter,
+                        )?;
+                    } else {
+                        return Err(PartialVMError::new(StatusCode::CONSTRAINT_NOT_SATISFIED)
+                            .finish(Location::Undefined));
+                    }
+                }
+                Err(err) => {
```

Comment:
Not familiar with verifier. we want to surface the error of init_module function that verifier catches when loading the function from module?

Reply:
> Not familiar with verifier. just to confirm I understand this change correctly. We want to surface the error of init_module function that verifier catches when loading the function from module?

the bottom line here and how explained in the comments and examples, is that there are situations where attempting to run the `init_module` generates a verifier error which was discarded, the init not run and the publish successful.
So all this is doing it to surface the error and fail the publishing and transaction. 
Does it make sense? 
thoughts?

Reply:
Make sense. Yes, surfacing these errors is helpful